### PR TITLE
Garfield++: Add the patch to fix the missing headers and Set the environment variable HEED_DATABASE

### PR DESCRIPTION
**Pull request by mirguest**
This PR will fix the issue in https://github.com/key4hep/key4hep-spack/issues/518

### CODE REVIEW

**Review by vvolkl — APPROVED** [review]

File: var/spack/repos/builtin/packages/garfieldpp/package.py [hunk 1]
```diff
@@ -17,9 +17,17 @@ class Garfieldpp(CMakePackage):
     tags = ["hep"]
 
     maintainers("mirguest")
-
+    # ###################### Patches ##########################
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/garfieldpp/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/garfieldpp/package.py [hunk 2]
```diff
@@ -17,9 +17,17 @@ class Garfieldpp(CMakePackage):
...
+    patch(
+        "https://gitlab.cern.ch/garfield/garfieldpp/-/commit/882c3023cfa89b45ca7a0c95ab1518454536e8e1.patch",
+        sha256="440bc8129c55168e6c45d39e4344911d48ddb13fd3f9ee05974b2ede46a23b93",
+        when="@4.0",
+    )
+
+    # ###################### Variants ##########################
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/garfieldpp/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/garfieldpp/package.py [hunk 3]
```diff
@@ -17,9 +17,17 @@ class Garfieldpp(CMakePackage):
...
     variant("examples", default=False, description="Build garfield examples")
 
+    # ###################### Versions ##########################
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/garfieldpp/package.py]:
```suggestion
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Minor request to remove unneeded comments

**Comment by mirguest** [comment]
> Minor request to remove unneeded comments

Thank you @alalazo . It's done. 

Tao

**Review by alalazo — APPROVED** [review]

**Pull request merged**